dana/programs/binarysearch.dana

```diff
@@ -0,0 +1,65 @@
+def main
+
+    def partition: arr as int [], low high as int
+        var pivot i j is int
+        pivot := arr[high]
+        i := low - 1
+        j := low
+        loop:
+            if j > high: break
+            if arr[j] < pivot:
+                i := i + 1
+                swap: arr[i], arr[j]
+            j := j + 1
+        swap: arr[i+1] arr[high]
+        return: (i+1)
+
+    def quicksort: arr as int [], low high as int
+        var pi is int
+        if low < high:
```

Comment:
I suspect that the indentation of this `if` is wrong. All three lines below it should be inside its `then` branch, or?